apps/site/src/app/apply/sections/Components/RadioSelect.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+
+import { useState, useEffect, useRef, forwardRef } from "react";
+
+interface RadioInputs {
+	name: string;
+	labelText: string;
+	IdentifierId: string;
+	values: Array<{ value: string; text: string }>;
+	containerClass: string;
+}
+
+interface IsChecked {
+	isChecked: boolean;
+	id: string;
+}
+
+const OtherInput = forwardRef<HTMLInputElement, IsChecked>(
+	({ isChecked, id }, ref) => (
+		<input
+			ref={ref}
+			type="text"
+			name={`${id}-other`}
+			className={
+				isChecked
+					? "border-b-2 p-1 h-6 border-black w-6/12"
+					: "border-b-2 p-1 h-6 border-black w-6/12 bg-transparent"
+			}
+			required={isChecked}
+			disabled={!isChecked}
+		/>
+	),
+);
+OtherInput.displayName = "OtherInput";
+
+export default function RadioSelect({
+	name,
+	labelText,
+	IdentifierId,
+	values,
+	containerClass,
+}: RadioInputs) {
+	const [isOtherChecked, setIsOtherChecked] = useState(false);
+	const otherRef = useRef<HTMLInputElement>(null);
+
+	useEffect(() => {
+		if (isOtherChecked) {
+			otherRef.current?.focus();
+		}
+	}, [isOtherChecked]);
+
+	return (
+		<div className={containerClass}>
+			<p className="m-0 text-lg mb-4">
+				{labelText} <span className="text-[#FF2222]">*</span>
+			</p>
+			<div className="w-10/12 flex flex-col gap-2">
+				{values.map((item, i) => {
+					const keyandId = `${IdentifierId}-${i}`;
+					if (item.value == "other") {
```

Comment:
Nitpick: We generally want to use strict equality (i.e. `===` instead of `==`) to prevent any unwanted behavior.

Reply:
Hmm, did eslint not pick this up?